### ISSUE TITLE
MAS Build got rejected  2.0.0-beta.2 using non-public framework

### ISSUE THREAD

**Issue by zinne** [electron/electron#12184]
Electron version: 2.0.0-beta.2
Operating system: macOS High Sierra 10.13.3
Actual behavior
Correctly packaged and signed app got rejected on the Mac AppStore. The current version 2.0.0-beta.2 didn't pass the review (app with electron 1.8.2 was OK)  with the following message:
```
Your app app links against the following non-public framework(s):

-‘/usr/lib/libSystem.B.dylib’ 
```
@jkleinsc is this also an issue of the beta build

**Comment by timfish** [comment]
are you sure you're not [hitting this](https://github.com/electron/electron/issues/12064#issuecomment-371225563)?

**Comment by MarshallOfSound** [comment]
This would be the issue solved in #12064

**Comment by zinne** [comment]
#12064 was my issue before. @jkleinsc told me it was fixed. And it works. This one is a new rejection issue... please reopen this issue. This is NOT the same as #12064

**Comment by MarshallOfSound** [comment]
Heh, sorry @zinne just scanned a list of things that were removed from the fixed app and a link to this was one of them.  May have jumped the gun 😆

**Comment by zinne** [comment]
if I ran
`$ otool -L ../Contents/Frameworks/Electron\ Framework.framework/Electron\ Framework`
it displays `/usr/lib/libSystem.B.dylib` under all other used libs...
Strange is that all the releases before were not rejected although the same dylib with the same path appears in the list on older versions of my app used electron 1.7.x. 
??? I have no idea... is Apple marking some libs as non public overnight?

**Comment by zcbenz** [comment]
@zinne The `libSystem.B.dylib` is implicitly linked for almost every macOS programs, for example `otool -L $(which ls)`, can you ask Apple more about the reason of the rejection?

**Comment by zinne** [comment]
@zcbenz yep I asked them... time to wait another 2 days :-/

**Comment by zinne** [comment]
@zcbenz this is the answer:
Hello,

Thank you for your response. To clarify, the following API from ’/usr/lib/libSystem.B.dylib' is not public:
‘sandbox_check’

Regards,

App Store Review

**Comment by sethlu** [comment]
It appears that the private API usage of `sandbox_check` is mentioned in https://github.com/electron/electron/issues/12064 as well. After doing some look up, it appears this method is probably only called in Chrome?

https://cs.chromium.org/chromium/src/sandbox/mac/seatbelt.cc?ssfr=1&type=cs&q=sandbox_check&l=67

**Comment by zcbenz** [comment]
@sethlu @zinne Thanks for the information, I'll patch the calls in Chromium.

**Comment by MarshallOfSound** [comment]
Fixed in #12264
